src/dmd/expression.d

```diff
@@ -4299,7 +4299,6 @@ extern (C++) final class VarExp : SymbolExp
         super(loc, TOK.variable, __traits(classInstanceSize, VarExp), var, hasOverloads);
         //printf("VarExp(this = %p, '%s', loc = %s)\n", this, var.toChars(), loc.toChars());
         //if (strcmp(var.ident.toChars(), "func") == 0) assert(0);
-        this.type = var.type;
```

Comment:
Is this really the correct fix?

Reply:
I'm not 100% sure, but if you search expressionsem.d for `if (e.type)` you will see it is a common idiom.  The expression classes don't seem to have a flag indicating whether or not semantic has been run, so it looks like the `type` property is (ab)used for that.

That being said, I don't see my `if (type)` check in the diff, so I may have neglected to add it.  It doesn't work at the moment anyway because there are other places in the code that don't call `expressionSemantic` on the `VarExp` class, so I need to submit a separate PR to remedy that.  I worked on it all day today, and it's proving to be quite difficult.

If I don't this, then a deprecation message is printed every type `VarExp.expressionSemantic` is run, and that's suboptimal.  I need some way to know whether or it has already been run, and the dominating idiom for that seems to be `if (e.type)` in the semantic visitor functions.